tests/Microsoft.Identity.Test.LabInfrastructure/LabAuthenticationHelper.cs

```diff
@@ -8,72 +8,103 @@
+            {
+                Debug.WriteLine("[LabAPI] Not on CI, using interactive browser/broker credential");
+                tokenCredential = GetAzureCredentialForDevBox();
+            }
+            else
+            {
+                Debug.WriteLine("[LabAPI] On CI, using ADO federation");
+                tokenCredential = GetAzureCredentialForCI();
+            }
+
+            return tokenCredential;
         }
 
-        public static async Task<AccessToken> GetLabAccessTokenAsync(string authority, string[] scopes, string clientId)
+        private static TokenCredential GetAzureCredentialForCI()
         {
-            AuthenticationResult authResult;
-            IConfidentialClientApplication confidentialApp;
-            X509Certificate2 cert;
+            // as per https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/samples/OtherCredentialSamples.md#authenticating-in-azure-pipelines-with-service-connections
 
-            var clientIdForCertAuth = String.IsNullOrEmpty(clientId) ? LabAccessConfidentialClientId : clientId;
+            string clientId = "4b7a4b0b-ecb2-409e-879a-1e21a15ddaf6"; // UAMI client ID
+            string tenantId = LabApiConstants.LabClientTenantId;
+            string serviceConnectionId = "6eeeb73d-37aa-4d78-83b7-728101b8bddd";
 
-            cert = CertificateHelper.FindCertificateByName(TestConstants.AutomationTestCertName);
-            if (cert == null)
-            {
-                throw new InvalidOperationException(
-                    "Test setup error - cannot find a certificate in the My store for KeyVault. This is available for Microsoft employees only.");
-            }
+            var pipelinesCredential = new AzurePipelinesCredential(
+                tenantId,
+                clientId,
+                serviceConnectionId,
+                Environment.GetEnvironmentVariable("SYSTEM_ACCESSTOKEN"),
+                new AzurePipelinesCredentialOptions()
+                {
+                    TokenCachePersistenceOptions = new TokenCachePersistenceOptions()
+                    {
+                        Name = "MSIDLabTokenCache",
+                        UnsafeAllowUnencryptedStorage = true // We generally use headless Linux, so cannot use LibSecret. This is the ~same level of protection as SSH keys.
+                    }
+                });
+
+            return pipelinesCredential;
+        }
 
-            confidentialApp = ConfidentialClientApplicationBuilder
-                .Create(clientIdForCertAuth)
-                .WithAuthority(new Uri(authority), true)
-                .WithCacheOptions(CacheOptions.EnableSharedCacheOptions)
-                .WithCertificate(cert, true)
-                .Build();
+        // TODO: test this on MacOs / Linux WSL
+        private static TokenCredential GetAzureCredentialForDevBox()
+        {
+            InteractiveBrowserCredential interactiveBrowserCredential = new InteractiveBrowserCredential(
```

Comment:
can we use `VisualStudioCredential` so it works even on a VPN'd dev machine? 

```cs
// Set the Key Vault URL
string keyVaultUrl = "https://msidlabs.vault.azure.net/";

// Create a new Visual Studio Credential
var credential = new VisualStudioCredential();

// Create a new SecretClient using the Visual Studio Credential
var client = new SecretClient(new Uri(keyVaultUrl), credential);

// Retrieve a secret from Azure Key Vault
KeyVaultSecret secret = client.GetSecret("msidlab1");

Console.WriteLine($"Secret: {secret.Value}");
```
this will use the Azure Service Authentication account 

![image](https://github.com/user-attachments/assets/2fc6bdf0-1b6e-4e2b-ace1-4023d88fc7e7)

Reply:
No, we can't use any of the DefaultAzureCredential options, because there is no option to change the clientID. If you use VSCredentail, it'll use VS clientID.

And VS clientID is not authorized to call MSIDlab.

I was thinking that if we were to directly call the KV that might work though. Maybe not VS client ID, but az cli or some tool might do it

Reply:
`VS clientID` is now authorized to call Lab API for an authorized user